### ISSUE TITLE
redundant_conversion: false positive

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#4254]
Here's an example where CONV is required but abaplint gives an error:

```abap
  TYPES ty_p TYPE p LENGTH 5 DECIMALS 2.
  DATA x TYPE ty_p VALUE '4.1'.
  DATA(y) = CONV ty_p( 2 * x ).  "<<< Redundant CONV expression (redundant_conversion)
```

Reason:

If you remove `CONV` and use `DATA(y) = 2 * x `, you get the following warning in ABAP:

<img width="571" height="99" alt="Image" src="https://github.com/user-attachments/assets/06175e1f-813b-49e5-a21d-1398450040ca" />

**Comment by larshp** [comment]
hm, yea, its one of the difficult ones
